druid/src/core.rs

```diff
@@ -81,6 +81,10 @@ pub struct WidgetState {
     /// `size` these constitute the child's layout rect.
     origin: Point,
     /// The origin of the parent in the window coordinate space;
+    //TODO: decide whether we should remove this. With this PR (https://github.com/linebender/druid/pull/2149)
+    // the position of a widget can change during event, but its global position only gets updated
+    // after layout while calling some_pod.layout_rect() gives you the correct local position every
+    // time.
     pub(crate) parent_window_origin: Point,
```

Comment:
The comment seems wrong though. It says that the global position only gets updated after layout, but it actually gets updated during `LifeCycle::ViewContextChanged`. So for example when I scroll, the event handler sets a new origin and then `LifeCycle::ViewContextChanged` gets sent from the window, but there is never any layout.

So to me it seems that `parent_window_origin` is stale only temporarily.
- `event()` happens, `set_origin()` gets called
    - `parent_window_origin` is now stale
- `LifeCycle::ViewContextChanged` gets sent, updates `parent_window_origin`
    - `parent_window_origin` is now synced again


Am I misunderstanding something? Because it seems to me right now, that the staleness is only during the specific event that triggers `set_origin()`, after `set_origin()` was called.

Reply:
My bad. Yes, when no widget needs layout LifeCycle::ViewContextChanged is sent directly after event and again after update if needed. Im not sure though if this is good. Its probably best to just send it once directly before paint.

> Am I misunderstanding something? Because it seems to me right now, that the staleness is only during the specific event that triggers set_origin(), after set_origin() was called.

Yeah, you are right. And the more i think about it, it seems unlikely, that in the same event / update cycle a widget changes the origin of an widget and another widget further down the hierarchy uses the global origin.

Reply:
My bad. Yes, when no widget needs layout LifeCycle::ViewContextChanged is sent directly after event and again after update if needed. Im not sure though if this is good. Its probably best to just send it once directly before paint.

> Am I misunderstanding something? Because it seems to me right now, that the staleness is only during the specific event that triggers set_origin(), after set_origin() was called.

Yeah, you are right. And the more i think about it, it seems unlikely, that in the same event / update cycle a widget changes the origin of an widget and another widget further down the hierarchy uses the global origin.

Reply:
I think that sending `LifeCycle::ViewContextChanged` quickly after event is fine, it reduces the chance of developers using stale data. The performance cost should be negligible.

Overall I think it's fine to keep `parent_window_origin` functioning as it is right now with this PR. Sure it could be improved in some way, but I think that can be proposed and discussed in separate PR if anyone has a desire for it. The problem isn't big enough to be required to solve here. Especially as removing it completely would remove some public API methods too.

What we should do is improve the documentation a bit around this issue. There are two target demographics if you will. Internal Druid developers and external Druid developers.

For the external ones, I think we should add a note to the docs of `set_origin()`.
```rust
    /// The changed origin won't be fully in effect until [`LifeCycle::ViewContextChanged`] has
    /// finished propagating. Specifically methods that depend on the widget's origin in relation
    /// to the window will return stale results during the period after calling `set_origin` but
    /// before [`LifeCycle::ViewContextChanged`] has finished propagating.
```

For the internal ones, let's change the comment of `parent_window_origin`.
```rust
    /// The origin of the parent in the window coordinate space.
    /// 
    /// Note that this value can be stale after `set_origin` was called
    /// but before `LifeCycle::ViewContextChanged` has fully propagated.
```

Reply:
I think that sending `LifeCycle::ViewContextChanged` quickly after event is fine, it reduces the chance of developers using stale data. The performance cost should be negligible.

Overall I think it's fine to keep `parent_window_origin` functioning as it is right now with this PR. Sure it could be improved in some way, but I think that can be proposed and discussed in separate PR if anyone has a desire for it. The problem isn't big enough to be required to solve here. Especially as removing it completely would remove some public API methods too.

What we should do is improve the documentation a bit around this issue. There are two target demographics if you will. Internal Druid developers and external Druid developers.

For the external ones, I think we should add a note to the docs of `set_origin()`.
```rust
    /// The changed origin won't be fully in effect until [`LifeCycle::ViewContextChanged`] has
    /// finished propagating. Specifically methods that depend on the widget's origin in relation
    /// to the window will return stale results during the period after calling `set_origin` but
    /// before [`LifeCycle::ViewContextChanged`] has finished propagating.
```

For the internal ones, let's change the comment of `parent_window_origin`.
```rust
    /// The origin of the parent in the window coordinate space.
    /// 
    /// Note that this value can be stale after `set_origin` was called
    /// but before `LifeCycle::ViewContextChanged` has fully propagated.
```